siriuspy/siriuspy/devices/dvf.py

```diff
@@ -385,22 +400,45 @@ def cmd_reset(self, timeout=None):
         self.acquisition_time = params.ACQUISITION_TIME_DEFAULT
         return True
 
-    def cmd_acquire_on(self, timeout=None):
+    def cmd_acquire_on(self, timeout=_default_timeout):
         """Tune IOC image acquisition on."""
         return self._set_and_wait('cam1:Acquire', 1, timeout=timeout)
 
-    def cmd_acquire_off(self, timeout=None):
+    def cmd_acquire_off(self, timeout=_default_timeout):
         """Tune IOC image acquisition off."""
         return self._set_and_wait('cam1:Acquire', 0, timeout=timeout)
 
-    def cmd_set_cam_roi(self, offsetx, offsety, width, height, timeout=None):
+    def cmd_cam_roi_set(self, offsetx, offsety, width, height, timeout=None):
         """Set cam image region of interest and reset aquisition."""
+        offsetx, offsety = int(offsetx), int(offsety)
+        width, height = int(width), int(height)
+        if offsetx + width > self.cam_max_sizex:
+            raise ValueError('Invalid offsetx and width combination!')
+        if offsety + height > self.cam_max_sizey:
+            raise ValueError('Invalid offsety and height combination!')
+
         self.cmd_acquire_off(timeout=timeout)
-        self.cam_offsetx = offsetx
-        self.cam_offsety = offsety
-        self.cam_width = width
-        self.cam_height = height
```

Comment:
I don't like much of this out-of-bounds checking duplication. I'd rather copying the current value `self.cam_width` and restoring it afterwards if anything doesn't match. This way only the `self.cam_{width,height}` setter deals with checking if the new value is valid or not.

Reply:
I reimplemented the method. see if you agree, @henriquesimoes !

Reply:
That seems better!